common/lib/client/auth.js

```diff
@@ -38,27 +38,25 @@ var Auth = (function() {
 		return JSON.stringify(c14nCapability);
 	}
 
+	function containsAuthInfo(options) {
+		var authMethods = ['authUrl', 'authCallback', 'token', 'tokenDetails', 'key'];
+		return (Utils.arrAny(authMethods, function(method) {
+			return (method in options);
+		}));
+	}
+
 	function Auth(client, options) {
 		this.client = client;
 		this.tokenParams = options.defaultTokenParams || {};
 
-		/* RSA7a4: if options.clientId is provided and is not
-		 * null, it overrides defaultTokenParams.clientId */
-		if(options.clientId) {
-			this.tokenParams.clientId = options.clientId;
-			this.clientId = options.clientId
-		}
-
 		/* decide default auth method */
 		var key = options.key;
 		if(key) {
 			if(!options.clientId && !options.useTokenAuth) {
```

Comment:
> Shouldn't this be checking if any token auth params are provided as well.

The logic of when to use basic vs token auth isn't something I changed here - ably-js has always used basic if a key is given as long as `useTokenAuth` isn't set. That said, looking at `RSA4` now, you're right, the logic not spec-compliant. I'll add a separate commit to fix that.

> Also clientId could be passed in with defaultTokenParams.clientId which would then be ignored and basic auth assumed.

I'm not sure I agree with this bit - should merely providing `defaultTokenParams` really force token auth used if it wouldn't otherwise have been used? That seems surprising. If someone wants an identified client they should pass a `clientId` in `clientOptions` directly